src/System.Device.Gpio/Interop/Unix/libgpiod/Interop.libgpiod.cs

```diff
@@ -5,54 +5,55 @@
 using System;
 using System.Device.Gpio;
 using System.Runtime.InteropServices;
+using NativeLong = System.IntPtr;
 
 internal partial class Interop
 {
-    private const string LibgpiodLibrary = "System.Device.Gpio.Native";
+    private const string LibgpiodLibrary = "libgpiod";
     internal static IntPtr InvalidHandleValue = new IntPtr(-1);
 
     /// <summary>
     /// Create a new gpiochip iterator.
     /// </summary>
     /// <returns>Pointer to a new chip iterator object or <see langword="null" /> if an error occurred.</returns>
```

Comment:
I don't think it can return `null`.  If the native function returns NULL that becomes a new `SafeChipIteratorHandle` whose handle value is 0 (and thus, presumably, IsInvalid returns true).

Reply:
I see, yeah the thing it returns is NULL:

```c
/**
 * @brief Create a new gpiochip iterator.
 * @return Pointer to a new chip iterator object or NULL if an error occurred.
 *
 * Internally this routine scans the /dev/ directory for GPIO chip device
 * files, opens them and stores their the handles until ::gpiod_chip_iter_free
 * or ::gpiod_chip_iter_free_noclose is called.
 */
struct gpiod_chip_iter *gpiod_chip_iter_new(void) GPIOD_API;
```

So I guess that I'll change the docs